### ISSUE TITLE
Documentation Fix: Document that rclone fallbacks to using the LOGNAME environment variable if USER not set

### ISSUE THREAD

**Issue by raphaelcohn** [rclone/rclone#5407]
#### The associated forum post URL from `https://forum.rclone.org`

N/A

#### What is your current rclone version (output from `rclone version`)?

```
rclone v1.55.1
- os/type: darwin
- os/arch: amd64
- go/version: go1.16.3
- go/linking: dynamic
- go/tags: cmount
```

#### What problem are you are trying to solve?

Ensuring documentation is accurate regarding environment variables.

#### How do you think rclone should be changed to solve that?

Wherever the documentation states that rclone defaults to using the `USER` environment variable, (eg the SFTP backend) it should be updated to note that it then falls back to using the `LOGNAME` environment variable if that is not set. This is done by the function `func CurrentUser()` in `env.go`. Actually, the scenario is more complex, as it uses Current user as a fallback before using `LOGNAME`.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by albertony** [comment]
The logic seems to be to return the first one found, from:
1. user.Current() from golang
   - On linux uses current `ùid` and looks it up in `/etc/passwd` first, fallback to `$USER` from environment
   - On windows uses Windows API syscall's
2. `USER` environment variable
3. `LOGNAME` environment variable

I would assume it rarely, if ever, gets past step 1... Have you experienced it actually reaching step 3?

Can only see CurrentUser being referenced from ftp and sftp backends, currently.

**Comment by raphaelcohn** [comment]
I understand the intent of the code; I just want rclone's documentation to accurately reflect what it's doing; the current SFTP documentation is incorrect. My copy of the rclone code seems to check `USER` before `user.Current()`, but I'm working from memory.

I actually don't mind the order, particularly - just want the documentation to match.

In a hardened environment `/etc/passwd` isn't always readable or present. Personally, I'm not much of a fan of inheriting important details from environment variables - and, in this case, whilst some ssh, sftp and ftp programs historically use the current logged in user's name, I'd argue it's actually a security flaw waiting to happen: running a raw sftp connection to as-yet-to-be-trusted ssh server (ie not yet validated as a host) with the current user's name seems to be asking for trouble.

**Comment by raphaelcohn** [comment]
One other place the docs are wrong is the base of <https://rclone.org/docs/> - there's no mention of `USER` or `LOGNAME`.

**Comment by albertony** [comment]
I see your point.

Regarding the reference to `$USER` in [ftp](https://rclone.org/ftp/#ftp-user), [sftp](https://rclone.org/sftp/#sftp-user) and [flags](https://rclone.org/flags/) section: These doc strings are generated, and are the same as shown in command line help text. When you run `rclone help backend sftp` you will see the actual username that will be used as default on your system. When the documentation is generated it forces the actual string `$USER` (and for reference to home path it uses the string `$HOME`), more like a placeholder (or what to call it). So when documentation mentions `$USER` it is not the actual environment variable, but "the username we are able to find".

As a first quick improvement, I can mention the use of `USER` and `LOGNAME` variables in the following section: 
https://rclone.org/docs/#other-environment-variables

Something like:
- `USER` and `LOGNAME` values are used as fallbacks for current username. The primary method for looking up username is OS-specific: Windows API on Windows, /etc/passwd on Unix systems. In the documentation the current username is simply referred to as `$USER`.

If we need to describe this in more detail, perhaps mentioning your concern, I think we need a separate chapter for it. Maybe in https://rclone.org/docs/, with links from ftp and sftp where it is used. I'm not sure if we need this? And also, if the reference to `$USER` mentioned in the beginning then should be something else to avoid indicating it is the actual environment variable, something like `<username>`. Think I need some more opinions up front here, to avoid wasting time.. @ncw?

**Comment by raphaelcohn** [comment]
I think that would be very useful, and enough for folks like me I think. I would just double check whether `user.Current()` uses real or effective user id, and note that. I don't know the function, but I'd hazard it almost certainly uses effective uid.

**Comment by albertony** [comment]
Thank you. I checked it, and it uses real user id. I will include that in the text.

**Comment by raphaelcohn** [comment]
Yes, it seems to. That's... interesting. But as long as one is aware, not a problem.
